content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,361 +35,643 @@ are covered here as well as the `System.Diagnostics` API.
+Install the following OpenTelemetry NuGet packages:
 
-#### ASP.NET Core
+[OpenTelemetry.Exporter.Console](https://www.nuget.org/packages/OpenTelemetry.Exporter.Console)
 
-To start tracing in an ASP.NET Core-based app, use the OpenTelemetry extensions
-for ASP.NET Core setup.
+[OpenTelemetry.Extensions.Hosting](https://www.nuget.org/packages/OpenTelemetry.Extensions.Hosting)
 
-First, ensure that you have the right packages:
 
 ```sh
-dotnet add package OpenTelemetry
-dotnet add package OpenTelemetry.Extensions.Hosting
 dotnet add package OpenTelemetry.Exporter.Console
+dotnet add package OpenTelemetry.Extensions.Hosting
 ```
 
-Then you can install the Instrumentation package
+For ASP.NET Core-based applications, also install the AspNetCore instrumentation package
+
+[OpenTelemetry.Instrumentation.AspNetCore](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNetCore)
 
 ```sh
-dotnet add package OpenTelemetry.Instrumentation.AspNetCore --prerelease
+dotnet add package OpenTelemetry.Instrumentation.AspNetCore
 ```
 
-Note that the `--prerelease` flag is required for all instrumentation packages
-because they are all dependent on naming conventions for attributes/labels
-(Semantic Conventions) that aren't yet classed as stable.
 
-Next, configure it in your ASP.NET Core startup routine where you have access to
-an `IServiceCollection`.
+### Initialize the SDK
+
+{{% alert title="Note" color="info" %}} If you’re instrumenting a library, you don't need to initialize the SDK. {{% /alert %}}
+
+It is important to configure an instance of the OpenTelemetry SDK as early as possible in your application.
+
+To initialize the OpenTelemetry SDK for an ASP.NET Core app like in the case of the example app, 
+update the content of the `Program.cs` file with the following code: 
 
 ```csharp
-using OpenTelemetry;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
 using OpenTelemetry.Resources;
 using OpenTelemetry.Trace;
 
-// Define some important constants and the activity source.
-// These can come from a config file, constants file, etc.
-var serviceName = "MyCompany.MyProduct.MyService";
+// Ideally, you will want this name to come from a config file, constants file, etc.
+var serviceName = "dice-server";
 var serviceVersion = "1.0.0";
 
 var builder = WebApplication.CreateBuilder(args);
 
-// Configure important OpenTelemetry settings, the console exporter
 builder.Services.AddOpenTelemetry()
-  .WithTracing(b =>
-  {
-      b
-      .AddSource(serviceName)
-      .ConfigureResource(resource =>
-          resource.AddService(
-            serviceName: serviceName,
-            serviceVersion: serviceVersion))
-      .AddAspNetCoreInstrumentation()
-      .AddConsoleExporter();
-  });
-```
+    .ConfigureResource(resource => resource.AddService(
+        serviceName: serviceName,
+        serviceVersion: serviceVersion))
+    .WithTracing(tracing => tracing
+        .AddSource(serviceName)
```

Comment:
AddSource should be using the name of the `ActivitySource`. In this example, it just happens that both service name and activitysource name are the same, but i'd recommend to explicitly make them same by calling something like AddSource(Instrumentation.ActivitySource.Name)